pkgs/applications/virtualization/virt-viewer/default.nix

```diff
@@ -22,8 +22,9 @@ stdenv.mkDerivation rec {
   buildInputs = [
     pkgconfig intltool glib libxml2 gtk3 gtkvnc gmp libgcrypt gnupg cyrus_sasl
     shared_mime_info libvirt libcap_ng yajl gsettings_desktop_schemas makeWrapper
-    xen numactl
-  ] ++ optionals spiceSupport [ spice_gtk spice_protocol libcap gdbm ];
+    numactl
+  ] ++ optionals spiceSupport [ spice_gtk spice_protocol libcap gdbm
+  ] ++ optional stdenv.is64bit xen;
```

Comment:
you are right that ```stdenv.system == "x86_64-linux"``` would better match that is written in xen's file.
or, better something like ```isWorkOn(stdenv.system, xen)``` which would read  ```xen```'s meta, if such function exist.